aten/src/ATen/native/Resize.cpp

```diff
@@ -21,15 +22,29 @@ Tensor& resize_cpu_(Tensor& self, IntArrayRef size) {
 // this function in native_functions.yaml
 Tensor& resize_as_sparse_(Tensor& self, const Tensor& src);
 
-Tensor& resize_as_(Tensor& self, const Tensor& the_template) {
+Tensor& resize_as_(
+    Tensor& self,
+    const Tensor& the_template,
+    c10::optional<MemoryFormat> optional_memory_format) {
+  auto memory_format =
+      optional_memory_format.value_or(MemoryFormat::Contiguous);
   if (self.is_sparse() && the_template.is_sparse()) {
+    TORCH_CHECK(
```

Comment:
@zou3519 suggestion doesn't seem done -- you can't do it without checking if a value is set and you are calling `value_or`.

Reply:
how about now

Reply:
yes, looks correct now.